### PR TITLE
Replace dashboard formsets

### LINKED ISSUE #1313: Split forms with formsets into separate views
To improve the consistency of the dashboard we should split the edit views that contain a form and a formset into two separate views. This problem occurs in:
- Attributes > Details > Edit attribute
    "Attribute values" formset should be removed from the edit view. Instead, in the attribute details view we should display a table presenting existing attribute values and a "Add" button redirecting to a new "Add/Edit attribute value view".
- Shipping methods > Details > Edit shipping method
   As in the previous case, "Price per country" should be replaced with a table and "Add" button on the "details" page.

The product details page may be used as a reference. Editing the product details and adding new variants follows the pattern that we want to introduce in the mentioned pages.

image

### PR DESCRIPTION
**Pull request by dominik-zeglen**
Resolves #1313

### CODE REVIEW

**Comment by koradon** [comment]
@elwoodxblues pls review this PR
Is it a part of 2017.11 milestone?

**Comment by elwoodxblues** [comment]
Yes, it is. I'll take a look.

File: templates/dashboard/product/product_attribute/form.html [hunk 1]
```diff
@@ -70,39 +70,39 @@
             <div class="row">
               {{ form.slug|materializecss }}
             </div>
-            <span class="card-title">
+            {% comment %}<span class="card-title">
```
Comment (elwoodxblues) [inline templates/dashboard/product/product_attribute/form.html]: Some commented out code left here and below.

File: saleor/dashboard/sites/views.py [hunk 2]
```diff
@@ -1,43 +1,92 @@
...
+    if key_pk:
+        key = get_object_or_404(AuthorizationKey, pk=key_pk)
+    else:
+        key = AuthorizationKey(site_settings_id=site_settings_pk)
+    form = AuthorizationKeyForm(request.POST or None, instance=key)
+    if form.is_valid():
+        key = form.save()
+        msg = pgettext_lazy(
+            'dashboard message',
+            'Updated country shipping price %s') % (key,) \
```
Comment (elwoodxblues) [inline saleor/dashboard/sites/views.py]: These messages here are from somewhere else :)

File: saleor/dashboard/product/views.py [hunk 3]
```diff
@@ -522,10 +519,54 @@ def attribute_delete(request, pk):
...
+@permission_required('product.edit_properties')
+def attribute_choice_value_delete(request, attribute_pk, value_pk):
+    value = get_object_or_404(AttributeChoiceValue, pk=value_pk)
+    if request.method == 'POST':
+        value.delete()
+        messages.success(
+            request,
+            pgettext_lazy(
+                'Dashboard message',
+                'Removed attribute choice value %s') % (value.name,))
```
Comment (elwoodxblues) [inline saleor/dashboard/product/views.py]: This message is inconsistent with messages in the "edit" view. I think better would be: "Removed attribute's value: %s".

File: saleor/dashboard/shipping/views.py [hunk 4]
```diff
@@ -78,8 +67,54 @@ def shipping_method_delete(request, pk):
...
+    else:
+        method = ShippingMethodCountry(shipping_method_id=shipping_method_pk)
+    form = ShippingMethodCountryForm(request.POST or None, instance=method)
+    if form.is_valid():
+        method = form.save()
+        msg = pgettext_lazy(
+            'Dashboard message',
+            'Updated country shipping price %s') % (method,) \
+            if country_pk else pgettext_lazy(
+            'Dashboard message', 'Added country shipping price %s') % (method,)
```
Comment (elwoodxblues) [inline saleor/dashboard/shipping/views.py]: I think a better message would be: "Added shipping price for %(country)".

File: saleor/dashboard/sites/views.py [hunk 5]
```diff
@@ -1,43 +1,92 @@
...
+
+    if site_settings_form.is_valid() and site_form.is_valid():
         site = site_form.save()
         site_settings_form.instance.site = site
         site_settings = site_settings_form.save()
-        formset.save()
-        messages.success(request, _('Updated site %s') % site_settings)
-        return redirect('dashboard:site-update', site_id=site_settings.id)
+        messages.success(request, pgettext_lazy(
+            'Dashboard message', 'Updated site %s') % site_settings)
```
Comment (elwoodxblues) [inline saleor/dashboard/sites/views.py]: "Updated site settings"

**Comment by koradon** [comment]
@dominik-zeglen we need to update field.html with 'is_hidden_input' check. Then we should update templates which are using workaround from latest commit.

**Comment by elwoodxblues** [comment]
I changed the rendering of the form and I think it works fine.

**Comment by koradon** [comment]
I updated rest of the forms and authorization key form template with this new rendering

**Pull request merged**